### ISSUE TITLE
Parsing Error: ClickHouse GRANT ALTER DELETE

### ISSUE THREAD

**Issue by mohaidoss** [sqlfluff/sqlfluff#7357]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

The parser fails when `Grant ALTER DELETE..` is used with clickhouse dialect.

> == [test.sql] FAIL                                                                                                                                                                                                                
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'GRANT
                       | ALTER DELETE ON db_name TO user_na...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
All Finished 📜 🎉!

### Expected Behaviour

This is a valid GRANT query in clickhouse

### Observed Behaviour

I noticed that the parser fails when there is a file with this GRANT present, it shouldn't

### How to reproduce

With the following `test.sql`file
```sql
GRANT ALTER DELETE ON db_name TO user_name;
```

running 
```bash
sqlfluff lint --dialect clickhouse test.sql
````


### Dialect

clickhouse

### Version

3.5.0

### Configuration

default file

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mohaidoss** [comment]
the parser result

> [L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'DelimiterGrammar'>..., <Delimited: [<Ref: 'StatementSegment'>]>, <AnyNumberOf:...]>"
[L:  1, P:  1]      |        word:                                                 'GRANT'
[L:  1, P:  6]      |        whitespace:                                           ' '
[L:  1, P:  7]      |        word:                                                 'ALTER'
[L:  1, P: 12]      |        whitespace:                                           ' '
[L:  1, P: 13]      |        word:                                                 'DELETE'
[L:  1, P: 19]      |        whitespace:                                           ' '
[L:  1, P: 20]      |        word:                                                 'ON'
[L:  1, P: 22]      |        whitespace:                                           ' '
[L:  1, P: 23]      |        word:                                                 'db_name'
[L:  1, P: 30]      |        whitespace:                                           ' '
[L:  1, P: 31]      |        word:                                                 'TO'
[L:  1, P: 33]      |        whitespace:                                           ' '
[L:  1, P: 34]      |        word:                                                 'user_name'
[L:  1, P: 43]      |        semicolon:                                            ';'
[L:  1, P: 44]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'GRANT ALTER DELETE
                       | ON db_name TO user_na...'

**Comment by WittierDinosaur** [comment]
The docs claim it's comma-delimited. Should it not be?
```sql
GRANT ALTER, DELETE ON db_name TO user_name;
```

**Comment by mohaidoss** [comment]
@WittierDinosaur it's different. The grant here is to be able to run this kind of query.

```sql
ALTER TABLE a DELETE WHERE col = 1
```
More details
https://clickhouse.com/docs/sql-reference/statements/alter/delete

Perhaps, same issue with ALTER UPDATE ?

**Comment by peterbud** [comment]
> The docs claim it's comma-delimited. Should it not be?
> 
> ```
> GRANT ALTER, DELETE ON db_name TO user_name;
> ```

@WittierDinosaur  `ALTER DELETE` is 1 privilege, so the GRANT uses comma separated privileges, just we don't have all privileges in the clickhouse grammar.
I believe we should extend/replace the ANSI 'AccessPermissionSegment' with all the defined privileges here:
https://clickhouse.com/docs/sql-reference/statements/grant
